### PR TITLE
woodpecker-*: 1.0.5 -> 2.2.2

### PR DESCRIPTION
**Pull request by ambroisie**
## Description of changes

Finally!

We're bypassing the pnpm packaging issue by using upstream's source tarball, which includes vendored dependencies and a pre-built frontend.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by ambroisie** [comment]
Need to test it on my server, and add a release note about the package upgrade (it is _technically_ breaking).

File: pkgs/development/tools/continuous-integration/woodpecker/common.nix [hunk 1]
```diff
@@ -26,7 +25,7 @@ in
   ldflags = [
     "-s"
     "-w"
-    "-X github.com/woodpecker-ci/woodpecker/version.Version=${version}"
+    "-X go.woodpecker-ci.org/woodpecker/version.Version=${version}"
```
Comment (ambroisie) [inline pkgs/development/tools/continuous-integration/woodpecker/common.nix:28]: It looks like the version isn't picked up correctly, my server thinks I'm running the dev version.

File: pkgs/development/tools/continuous-integration/woodpecker/update.sh [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #!/usr/bin/env nix-shell
-#!nix-shell -i bash -p nix wget prefetch-yarn-deps nix-prefetch-github jq nix-prefetch pnpm-lock-export
+#!nix-shell -i bash -p nix wget jq nix-prefetch pnpm-lock-export
```
Comment (ambroisie) [inline pkgs/development/tools/continuous-integration/woodpecker/update.sh:2]:
```suggestion
#!nix-shell -i bash -p nix wget jq nix-prefetch
```

**Review by adamcstephens — APPROVED** [review]
Result of `nixpkgs-review pr 284109` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>woodpecker-agent</li>
    <li>woodpecker-cli</li>
    <li>woodpecker-server</li>
  </ul>
</details>

-----

Result of `nixpkgs-review pr 284109` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>woodpecker-agent</li>
    <li>woodpecker-cli</li>
    <li>woodpecker-server</li>
  </ul>
</details>

-----

Result of `nixpkgs-review pr 284109` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>woodpecker-agent</li>
    <li>woodpecker-cli</li>
    <li>woodpecker-server</li>
  </ul>
</details>

-----

Result of `nixpkgs-review pr 284109` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>woodpecker-agent</li>
    <li>woodpecker-cli</li>
    <li>woodpecker-server</li>
  </ul>
</details>

**Comment by ambroisie** [comment]
Not sure what's going on with ofborg, but the failure looks to be unrelated.

**Comment by adamcstephens** [comment]
@ofborg eval

**Comment by Janik-Haag** [comment]
Uh great work, we really need some pnpm tooling for nix :(

**Comment by ambroisie** [comment]
Yeah

I don't know much about JS at all unfortunately, otherwise I'd have tried looking into it already.

**Pull request merged**